php/installer/Installer.class.inc

```diff
@@ -185,21 +186,30 @@ class Installer
             return false;
         }
         $DB = Database::singleton();
-        if ($DB->connectAndCreate(
-            $values['dbname'],
-            $values['dbadminuser'],
-            $values['dbadminpassword'],
-            $values['dbhost']
-        ) == false
-        ) {
+        try {
+            if ($DB->connectAndCreate(
+                $values['dbname'],
+                $values['dbadminuser'],
+                $values['dbadminpassword'],
+                $values['dbhost']
+            ) == false
+            ) {
+                if (!empty($DB->lastErr)) {
```

Comment:
This looks like it does the same thing as the code below. Isn't the try...catch the only part that's needed here?

Reply:
You're right, will remove.